src/context/DataContext.tsx

```diff
@@ -110,126 +119,106 @@ export const DataProvider: React.FC<DataProviderProps> = ({ children }) => {
-
     if (sortParams) {
       url += `&${sortParams}`
     }
-
     const filterString = buildFilterParams(filters)
     if (filterString) {
       url += `&${filterString}`
     }
-
     if (searchTerm) {
       url += `&search=${encodeURIComponent(searchTerm)}`
     }
-
     return url
   }, [currentPage, pageSize, sortParams, filters, searchTerm])
 
   const fetchData = useCallback(async () => {
+    const isDefaultView =
+      (!searchTerm || searchTerm.trim() === '') &&
+      (Object.keys(filters).length === 0 ||
+        Object.values(filters).every((filter: any) => !filter || !filter.value))
     setLoading(true)
+    if (isDefaultView && !metricsLoaded) {
+      setLoadingTotalNodes(true)
+    }
     try {
       const response = await axios.get(fetchUrl)
-
-      let sanitizedData: NodeData[] = []
-      for (let index = 0; index < response.data.nodes.length; index++) {
-        const element = response.data.nodes[index]
-
-        console.groupEnd()
-
-        sanitizedData.push({
-          ...element._source,
-          index: (currentPage - 1) * pageSize + index + 1
-        })
-      }
-
-      console.groupEnd()
-
+      const sanitizedData = response.data.nodes.map((element: any, index: number) => ({
+        ...element._source,
+        index: (currentPage - 1) * pageSize + index + 1
+      }))
       setData(sanitizedData)
       setTotalItems(response.data.pagination.totalItems)
       setNextSearchAfter(response.data.pagination.nextSearchAfter)
-      setTotalNodes(response.data.pagination.totalItems)
-      // setTotalEligibleNodes(response.data.totalEligibleNodes)
+      if (isDefaultView && !metricsLoaded) {
+        setTotalNodes(response.data.pagination.totalItems)
+      }
     } catch (err) {
       console.error('Error fetching data:', err)
       setError(err)
     } finally {
       setLoading(false)
+      if (isDefaultView && !metricsLoaded) {
+        setLoadingTotalNodes(false)
+      }
     }
-  }, [currentPage, fetchUrl, pageSize])
+  }, [currentPage, fetchUrl, pageSize, searchTerm, filters, metricsLoaded])
 
   const getTotalEligible = useCallback(async () => {
-    setLoading(true)
+    if (!metricsLoaded) setLoadingTotalEligible(true)
     const date = Date.now()
     const oneWeekInMs = 7 * 24 * 60 * 60 * 1000
+    const oneWeekAgo = Math.floor(new Date(date - oneWeekInMs).getTime() / 1000)
     try {
-      const oneWeekAgo = Math.floor(new Date(date - oneWeekInMs).getTime() / 1000)
       const response = await axios.get(
         `https://analytics.nodes.oceanprotocol.com/summary?date=${oneWeekAgo}`
       )
-
       setTotalEligibleNodes(response.data.numberOfRows)
     } catch (err) {
       console.error('Error total eligible nodes data:', err)
-      const twoWeekAgo = Math.floor(new Date(date - 2 * oneWeekInMs).getTime() / 1000)
-      const response = await axios.get(
-        `https://analytics.nodes.oceanprotocol.com/summary?date=${twoWeekAgo}`
```

Comment:
we still need to have this logic, so basically 
we need to be able to check the rewards for one 2 weeks because it possible at a time that rewards were not yet distributed for current week (aka one week ago) and we need to go 2 weeks prior to get the number of nodes that were eligbile for rewards